File: Examples/CSharp/Programming-Documents/Bookmarks/BookmarkNameAndText.cs
using System;
using System.IO;
using System.Reflection;
using Aspose.Words.Tables;
using Aspose.Words;

namespace Aspose.Words.Examples.CSharp.Programming_Documents.Bookmarks
{
    class BookmarkNameAndText
    {
        public static void Run()
        {
            //ExStart:BookmarkNameAndText
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_WorkingWithBookmarks();

            Document doc = new Document(dataDir + "Bookmark.doc");

            // Use the indexer of the Bookmarks collection to obtain the desired bookmark.
            Bookmark bookmark = doc.Range.Bookmarks["MyBookmark"];

            // Get the name and text of the bookmark.
            string name = bookmark.Name;
            string text = bookmark.Text;

            // Set the name and text of the bookmark.
            bookmark.Name = "RenamedBookmark";
            bookmark.Text = "This is a new bookmarked text.";
            //ExEnd:BookmarkNameAndText
            Console.WriteLine("\nBookmark text and name get and set successfully." + dataDir);
        }
        
    }
}

File: Examples/VisualBasic/VisualBasic/Programming-Documents/Bookmarks/BookmarkNameAndText.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.IO
Imports Aspose.Words
Public Class BookmarkNameAndText
    Public Shared Sub Run()
        ' ExStart:BookmarkNameAndText
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_WorkingWithBookmarks()

        Dim doc As New Document(dataDir & Convert.ToString("Bookmark.doc"))

        ' Use the indexer of the Bookmarks collection to obtain the desired bookmark.
        Dim bookmark As Bookmark = doc.Range.Bookmarks("MyBookmark")

        ' Get the name and text of the bookmark.
        Dim name As String = bookmark.Name
        Dim text As String = bookmark.Text

        ' Set the name and text of the bookmark.
        bookmark.Name = "RenamedBookmark"
        bookmark.Text = "This is a new bookmarked text."
        ' ExEnd:BookmarkNameAndText
        Console.WriteLine(Convert.ToString(vbLf & "Bookmark text and name get and set successfully.") & dataDir)
    End Sub
End Class
